internal/test/daemon/daemon.go

```diff
@@ -278,41 +278,46 @@ func (d *Daemon) StartWithLogFile(out *os.File, providedArgs ...string) error {
 
 	d.Wait = wait
 
+	clientConfig, err := d.getClientConfig()
+	if err != nil {
+		return err
+	}
+	client := &http.Client{
+		Transport: clientConfig.transport,
+	}
+
+	req, err := http.NewRequest("GET", "/_ping", nil)
+	if err != nil {
+		return errors.Wrapf(err, "[%s] could not create new request", d.id)
+	}
+	req.URL.Host = clientConfig.addr
+	req.URL.Scheme = clientConfig.scheme
+
 	ticker := time.NewTicker(500 * time.Millisecond)
 	defer ticker.Stop()
 	tick := ticker.C
 
+	timeout := time.NewTimer(60 * time.Second) // timeout for the whole loop
```

Comment:
I had to bump up this timeout because before the client was just waiting forever for a ping to complete and the (global) timeout didn't even matter in this case.

With the per-request timeout the global timeout actually has an effect on these cases and was causing failures... specifically testing for daemon startup with (on purpose) missing plugins, which takes ~30s when due to plugin connection retries during daemon initialization.